Rom.py

```diff
@@ -386,3 +403,30 @@ def free_space(self, size: Optional[int] = None) -> int:
                 return next(filter(lambda f: f[0] >= size, free_space))[1]
             except StopIteration:
                 raise Exception(f"Not enough free space in ROM to fit a file of size {size}. Largest region of free space available: {free_space[-1][0]}.")
+
+class OverlayEntry:
+    def __init__(self, vrom_start, vrom_end, vram_start, vram_end) -> None:
+        self.vrom_start = vrom_start
+        self.vrom_end = vrom_end
+        self.vram_start = vram_start
+        self.vram_end = vram_end
+
+class OverlayTable:
+    def read_overlay_table(rom: Rom, ovl_table_start: int, offset: int, entry_size: int, num_entries: int) -> None:
+        overlay_entries: list[OverlayEntry] = []
+        # Read the overlay table from the ROM
+        for i in range(0, num_entries):
+            entry_bytes = rom.read_bytes(ovl_table_start + i * entry_size, entry_size)
+            vrom_start = int.from_bytes(entry_bytes[offset + 0:offset +4], 'big')
+            vrom_end = int.from_bytes(entry_bytes[offset + 4:offset +8], 'big')
+            vram_start = int.from_bytes(entry_bytes[offset + 8:offset +12], 'big')
+            vram_end = int.from_bytes(entry_bytes[offset + 12:offset +16], 'big')
+            overlay_entries.append(OverlayEntry(vrom_start, vrom_end, vram_start, vram_end))
+        return overlay_entries
+
+    def VRAM_2_VROM(overlay_entries: list[OverlayEntry], vram_address: int) -> int:
+        # Loop through overlay table and find the entry containing the address
+        for overlay_entry in overlay_entries:
+            if (overlay_entry.vram_start <= vram_address) and (overlay_entry.vram_end > vram_address):
```

Comment:
```suggestion
            if overlay_entry.vram_start <= vram_address and overlay_entry.vram_end > vram_address:
```

Reply:
I'll let rob look at this one since this would have been in his commit.

Reply:
That's fine